pkg/releasetesting/test_suite.go

```diff
@@ -61,89 +61,97 @@ func NewTestSuite(rel *release.Release) (*TestSuite, error) {
 				return err
 			}
 		}
 
 		resourceCleanExit := true
 		status := api.PodUnknown
 		if resourceCreated {
-			status, err = t.getTestPodStatus(test, env)
+			status, err = env.getTestPodStatus(test)
 			if err != nil {
 				resourceCleanExit = false
-				if streamErr := streamUnknown(test.result.Name, test.result.Info, env.Stream); streamErr != nil {
+				if streamErr := env.streamUnknown(test.result.Name, test.result.Info); streamErr != nil {
 					return streamErr
 				}
 			}
 		}
 
-		if resourceCreated && resourceCleanExit && status == api.PodSucceeded {
-			test.result.Status = release.TestRun_SUCCESS
-			if streamErr := streamSuccess(test.result.Name, env.Stream); streamErr != nil {
-				return streamErr
+		if resourceCreated && resourceCleanExit {
+			if err := test.assignTestResult(status); err != nil {
+				return err
 			}
-		} else if resourceCreated && resourceCleanExit && status == api.PodFailed {
-			test.result.Status = release.TestRun_FAILURE
-			if streamErr := streamFailed(test.result.Name, env.Namespace, env.Stream); streamErr != nil {
+
+			if err := env.streamResult(test.result); err != nil {
 				return err
 			}
 		}
 
 		test.result.CompletedAt = timeconv.Now()
-		t.Results = append(t.Results, test.result)
+		ts.Results = append(ts.Results, test.result)
 	}
 
-	t.CompletedAt = timeconv.Now()
+	ts.CompletedAt = timeconv.Now()
 	return nil
 }
 
-// NOTE: may want to move this function to pkg/tiller in the future
-func filterHooksForTestHooks(hooks []*release.Hook, releaseName string) ([]*release.Hook, error) {
-	testHooks := []*release.Hook{}
-	notFoundErr := fmt.Errorf("no tests found for release %s", releaseName)
-
-	if len(hooks) == 0 {
-		return nil, notFoundErr
-	}
-
-	for _, h := range hooks {
-		for _, e := range h.Events {
-			if e == release.Hook_RELEASE_TEST_SUCCESS {
-				testHooks = append(testHooks, h)
-				continue
-			}
+func (t *test) assignTestResult(podStatus api.PodPhase) error {
+	switch podStatus {
+	case api.PodSucceeded:
+		if t.expectedSuccess == true {
+			t.result.Status = release.TestRun_SUCCESS
+		} else {
+			t.result.Status = release.TestRun_FAILURE
+		}
+	case api.PodFailed:
+		if t.expectedSuccess == false {
+			fmt.Println("expected to be here")
+			t.result.Status = release.TestRun_SUCCESS
+		} else {
+			fmt.Printf("expected success: %v", t.expectedSuccess)
+			t.result.Status = release.TestRun_FAILURE
 		}
+	default:
+		t.result.Status = release.TestRun_UNKNOWN
 	}
 
-	if len(testHooks) == 0 {
-		return nil, notFoundErr
-	}
+	return nil
+}
 
-	return testHooks, nil
+func expectedSuccess(hookTypes []string) (bool, error) {
+	for _, hookType := range hookTypes {
+		hookType = strings.ToLower(strings.TrimSpace(hookType))
+		fmt.Println("hook type " + hookType)
+		if hookType == hooks.ReleaseTestSuccess {
+			return true, nil
+		} else if hookType == hooks.ReleaseTestFailure {
+			return false, nil
+		}
+	}
+	return false, errors.New("No test-success or test-failure hook found")
```

Comment:
maybe format in actual hook consts in case hook values change in future? `errors.New(fmt.Sprintf("No %s or %s hook found", hooks.ReleaseTestSuccess, hooks.ReleaseTestFailure))`

Alternatively, perhaps create a specific `ErrUnknownTestHook` error that can be thrown?